src/snowflake/cli/api/project/schemas/native_app/application.py

```diff
@@ -23,6 +23,12 @@
 )
 
 
+class ApplicationPostDeployHook(UpdatableModel):
```

Comment:
have you tried this with multiple options to see if it works when we expand it in the future?

Reply:
you might have to add @root_validator to ensure that one and only one of the attributes is defined.

Reply:
Yes, that's a good suggestion. I think we should add this once there's actually a second attribute, because now it would be an unreachable code path.

Tried to add a second attribute locally and verified it works, including the validation:

```python
class ApplicationPostDeployHook(UpdatableModel):
    sql_script: Optional[str] = Field(
        title="SQL file path relative to the project root", default=None
    )
    grant: Optional[str] = Field(
        title="Grant a permission", default=None
    )

    @model_validator(mode='before')
    @classmethod
    def ensure_exactly_one_hook_type(cls, hook):
        supported_keys = cls.model_fields.keys()
        actual_keys = hook.keys()
        previously_found_key = None
        for key in actual_keys:
            if key in supported_keys:
                if previously_found_key:
                    raise ValueError(f"Only one of the following keys can be specified: {supported_keys}")
                else:
                    previously_found_key = True
        if not previously_found_key:
            raise ValueError(f"One of the following keys must be specified: {supported_keys}")

        return hook
```

```yml
# Works
post_deploy: 
  - sql_script: scripts/post_install.sql

# Works
post_deploy: 
  - grant: grant

# Error
post_deploy: 
  - sql_script: scripts/post_install.sql
  - grant: grant

# Error
post_deploy: 
  - something: else
```

Reply:
But you can use this to ensure there is exactly one attribute in use. In this case, if someone doesn't enter one attribute (sql_script for example), it should fail (even though it says optional).

Reply:
Good suggestion. Added custom validation to catch these 2 cases:
```yml
post_deploy: 
  - {}
```
```yml
post_deploy: 
  -
```